File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    answer = []
    
    soopo1 = [1, 2, 3, 4, 5]
    soopo2 = [2, 1, 2, 3, 2, 4, 2, 5]
    soopo3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    
    answer1 = [answers[i:i+len(soopo1)] for i in range(0, len(answers), len(soopo1))]
    cnt1 = 0
    for ans in answer1:
        for i in range(len(ans)):
            if ans[i] == soopo1[i]:
                cnt1 += 1
    answer.append(cnt1)
    
    answer2 = [answers[i:i+len(soopo2)] for i in range(0, len(answers), len(soopo2))]
    cnt2 = 0
    for ans in answer2:
        for i in range(len(ans)):
            if ans[i] == soopo2[i]:
                cnt2 += 1
    answer.append(cnt2)
    
    answer3 = [answers[i:i+len(soopo3)]for i in range(0, len(answers), len(soopo3))]
    cnt3 = 0
    for ans in answer3:
        for i in range(len(ans)):
            if ans[i] == soopo3[i]:
                cnt3 += 1
    answer.append(cnt3)
    
    result = []
    tmp = max(answer)
    for i in range(len(answer)):
        if answer[i] == tmp:
            result.append(i + 1)
    
    result.sort()
    
    return result

File: 프로그래머스/1/42840. 모의고사/모의고사.js
function solution(answers) {
    var answer = []
    var correct_cnt = [0, 0, 0];
    
    var soopo1 = [1, 2, 3, 4, 5]
    var soopo2 = [2, 1, 2, 3, 2, 4, 2, 5]
    var soopo3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    
    var answer_key1 = []
    var answer_key2 = []
    var answer_key3 = []
    for (let i = 0; i < answers.length; i += soopo1.length) {
        answer_key1.push(...soopo1)
        answer_key2.push(...soopo2)
        answer_key3.push(...soopo3)
    }
    
    for (let i = 0; i < answers.length; i++) {
        if (answer_key1[i] === answers[i]) correct_cnt[0]++
        if (answer_key2[i] === answers[i]) correct_cnt[1]++
        if (answer_key3[i] === answers[i]) correct_cnt[2]++
    }
    
    max_cnt = Math.max(...correct_cnt)
    for (let i = 0; i < correct_cnt.length; i++) {
        if (correct_cnt[i] === max_cnt) answer.push(i + 1)
    }
    
    return answer;
}